rs/protobuf/src/gen/types/state.queues.v1.rs

```diff
@@ -288,6 +290,15 @@ pub mod canister_queues {
         #[prost(message, optional, tag = "3")]
         pub output_queue: ::core::option::Option<super::CanisterQueue>,
     }
+    /// The callback ID of a response that was shed while in an input queue.
+    #[allow(clippy::derive_partial_eq_without_eq)]
```

Comment:
The knee-jerk answer to this is that `prost-build` will automatically add `#[derive(PartialEq)]` but not `#[derive(Eq)]` to the structs it generates. See [this `prost` issue](https://github.com/tokio-rs/prost/issues/661) for context.

Going deeper than that, my guess is that this is for the exact reason described in the [`PartialEq` documentation](https://doc.rust-lang.org/std/cmp/trait.PartialEq.html):
> For example, in floating point numbers `NaN != NaN`, so floating point types implement `PartialEq` but not `Eq`.

Meaning that we could likely go the other way with our replicated state protos (add a `#[derive(Eq)]` instead of the `#allow`), because we never have floats in them. But then we'd need other safety checks in place. And I don't think we'd gain much from it, if anything.

Reply:
Thanks. Makes sense.